app/controllers/UploadProxyController.scala

```diff
@@ -80,6 +114,41 @@ private object UploadProxyController {
 
   private val KeyName = "key"
 
+  object TemporaryFilePart {
+    private val AdoptedFileSuffix = ".out"
+
+    def mapToFailuresAndSuccesses[A, B](fileParts: Seq[FilePart[A]])
```

Comment:
feels like this could be simplified. Maybe do the map in the client, and just have a utility for the collection of the results?
e.g. `TemporaryFilePart.partitionTry(fileParts.map { ... })`

Reply:
worth using `collect` to avoid trusting `.failed`/`.get` to the previous partitioning by `isFailure` (I assume there are never that many, so traversing twice isn't an issue)?
```scala
( results.collect {  case Failure(e) => s"$err..." },
  results.collect { case Success(s) => s
)
```

it doesn't look so overwhelming, that maybe it could be inlined to make the overall processing of the results easier to follow?